_posts/2020-09-16-#19845.md

```diff
@@ -0,0 +1,139 @@
+---
+layout: pr
+date: 2020-09-16
+title: "Net: CNetAddr: add support to (un)serialize as ADDRv2"
+pr: 19845
+authors: [vasild]
+components: ["p2p"]
+host: jonatack
+status: upcoming
+commit:
+---
+
+## Notes
+
+### Motivation
+
+- Today's PR 19845 is part of [PR
+  19031](https://github.com/bitcoin/bitcoin/pull/19031), a proposed
+  implementation of the
+  [BIP155](https://github.com/bitcoin/bips/blob/9286b5254317d9e73fb25c5f0acd2b2d9937843e/bip-0155.mediawiki)
+  `addrv2` message.
+
+- `addrv2` is a new p2p message format proposed in early 2019 by Wladimir J. van
+  der Laan to support networks that have longer endpoint addresses than fit in
+  the 128 bits/16 bytes of the current `addr` message.
+
+- This work is urgent because Tor v2 was deprecated yesterday (September 15,
+  2020) with 0.4.4.x and will be obsoleted in 0.4.6.x (July 15, 2021). Tor v2 is
+  expected to be completely disabled in Tor client stable versions on October
+  15, 2021 as [announced
+  here](https://blog.torproject.org/v2-deprecation-timeline): "From today (July
+  2nd, 2020), the Internet has around 16 months to migrate from onion services
+  v2 to v3 once and for all."
+
+- Why is Tor v2 EOL? According to [this Tor mailing list
+  post](https://lists.torproject.org/pipermail/tor-dev/2020-June/014365.html):
+  "To very quickly summarize why we are deprecating, in one word: Safety. Onion
+  service v2 uses RSA1024 and 80 bit SHA1 (truncated) addresses. It also still
+  uses the TAP handshake which has been entirely removed from Tor for many years
+  now _except_ v2 services. Its simplistic directory system exposes it to a
+  variety of enumeration and location-prediction attacks that give HSDir relays
+  too much power to enumerate or even block v2 services. Finally, v2 services
+  are not being developed nor maintained anymore. Only the most severe security
+  issues are being addressed."
+
+- A [previous review club meeting discussing the parent PR 19031](/19031)
+  provides more background and context.
+
+### Networks newly supported by PR 19845
```

Comment:
suggestion by @vasild here, as support isn't fully operational yet for these:

"New addresses recognized by PR 19845"

Reply:
Done in #256.